src/main/java/com/thealgorithms/maths/SecondMinMax.java

```diff
@@ -0,0 +1,46 @@
+package com.thealgorithms.maths;
+
+public class SecondMinMax {
+
+    /**
+     * @brief Finds the Second minimum / maximum value from the array
+     * @param int array
+     * @exception IllegalArgumentException if input array is empty or length 1 also if all elements are same
+     * @return the second minimum / maximum value from the input array
+     * @author Bharath Sanjeevi ( https://github.com/BharathSanjeeviT )
+     */
+
+    public static int findSecondMin(int[] arr) {
+        int secMin = Integer.MAX_VALUE, min = Integer.MAX_VALUE;
+        if (arr.length == 0)
+            throw new IllegalArgumentException("Cant find Second Minimum for empty array");
+        else if (arr.length == 1)
+            throw new IllegalArgumentException("Cant find Second Minimum for single element");
```

Comment:
This logic is repeated in both methods - maybe you can abstract it to some private method like `checkInput(...)`.

Reply:
Sure will change this 👍